src/mongocxx/exception/base.hpp

```diff
@@ -30,33 +27,50 @@ namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace exception {
 
-using error_and_code_type = std::tuple<std::string, std::int32_t>;
-
+///
+/// A class to be used as the base class for all mongocxx exceptions.
+///
 class MONGOCXX_API base : virtual public std::exception {
```

Comment:
I think the benefit of having a common base class (which may itself inherit from std::system_error) is that all the exceptions need to hold an optional<document::value> of the raw server error if it is available